skopt/parameter.py

```diff
@@ -38,178 +37,156 @@ def inverse_transform(self, values):
+        * `high` [float]:
+            Upper bound of the parameter. Inclusive
         """
         self._low = low
         self._high = high
-
-        if transformer == 'identity':
-            self.transformer = Identity()
-        elif isinstance(transformer, TransformerMixin):
-            self.transformer = transformer
-        else:
-            raise RuntimeError('%s is not a valid transformer.'%transformer)
-
-        if prior == 'uniform':
-            self._rvs = randint(self._low, self._high)
-        elif isinstance(prior, rv_frozen):
-            self._rvs = prior
+        self._rvs = randint(self._low, self._high + 1)
+        self.transformer = Identity()
 
     def rvs(self, n_samples=None, random_state=None):
         return self._rvs.rvs(size=n_samples, random_state=random_state)
 
 
 class Categorical(Distribution):
-    def __init__(self, *categories, prior=None, transformer='labels'):
+    def __init__(self, *categories, prior=None):
         """Search space dimension that can take on categorical values.
 
         Parameters
         ----------
         *categories :
             sequence of possible categories
 
-        prior : array-like, length `len(categories)`, default None
+        * `prior` [array-like, shape=(categories,), default None]:
             Prior probabilities for each category. By default all categories
             are equally likely.
-
-        transformer: string or fitted TransformerMixin, default 'labels'
-            Transformer to convert between original and warped search space.
-            Parameter values are always transformed before being handed to the
-            optimizer. Defaults to `LabelEncoder`.
         """
-        self.categories = categories
-
-        if transformer == 'labels':
-            self.transformer = LabelEncoder()
-            self.transformer.fit(self.categories)
-        elif transformer == 'onehot':
-            # XXX Need a chain of transformers
-            self.transformer = OneHotEncoder()
-            self.transformer.fit(self.categories)
-        elif isinstance(transformer, TransformerMixin):
-            self.transformer = transformer
-        else:
-            raise RuntimeError('%s is not a valid transformer.'%transformer)
-
+        self.categories = np.asarray(categories)
+        self.transformer = CategoricalEncoder()
+        self.transformer.fit(self.categories)
         if prior is None:
-            prior = np.tile(1./len(self.categories), len(self.categories))
+            prior = np.tile(1. / len(self.categories), len(self.categories))
         self._rvs = rv_discrete(values=(range(len(self.categories)), prior))
 
     def rvs(self, n_samples=None, random_state=None):
         choices = self._rvs.rvs(size=n_samples, random_state=random_state)
-        if n_samples is None:
-            return self.categories[choices]
-        else:
-            return [self.categories[choice] for choice in choices]
+        return self.categories[choices]
```

Comment:
@betatim The inverse transform is actually done here. Might be better indeed to move this step (from integer to category) within an inverse_transform of LabelEncoder. In any case, this will eventually be required when presenting the solution to the user.